lib/Compilation.js

```diff
@@ -869,7 +869,7 @@ class Compilation {
 			currentProfile.markRestoringStart();
 		}
 
-		const cacheName = `${this.compilerPath}/module/${identifier}`;
+		const cacheName = `${this.compilerPath}!module!${identifier}`;
```

Comment:
Afaik `|` should also be possible as separator and would make more sense here, since `!` is usually used for loaders.